src/main/java/Duke.java

```diff
@@ -1,10 +1,186 @@
+        System.out.println("Got it. I've added this task:");
+        System.out.println(taskList[taskCount].toString());
+        taskCount++;
+        System.out.println("Now you have " + taskCount + " tasks in the list");
+    }
+
+
+    //add deadline task for level-4
+    public static void addDeadline(String[] inputComponent) {
+        String description = "";
+        String by = "";
+        int state = 0;//transition from "description" to "by" string
+
+
+        for (int i = 1; i < inputComponent.length; i++) {
+            if (inputComponent[i].equals("/by")) {
+                state = 1;
+            } else {
+                if (state == 1) {
+                    by += inputComponent[i];
+                    by += " ";
+                } else {
+                    description += inputComponent[i];
+                    description += " ";
+                }
+            }
+        }
+
+
+        taskList[taskCount] = new Deadline(description.trim(), by.trim());
+        System.out.println("Got it. I've added this task:");
+        System.out.println(taskList[taskCount].toString());
+        taskCount++;
+        System.out.println("Now you have " + taskCount + " tasks in the list");
+    }
+
+
+    //add event task for level-4
+    public static void addEvent(String[] inputComponent) {
+        String description = "";
+        String from = "";
+        String to = "";
+        int state = 0;//transition from "description" to "from" to "to" string
+
+
+        for (int i = 1; i < inputComponent.length; i++) {
+            if (inputComponent[i].equals("/from")) {
+                state = 1;
+            } else if (inputComponent[i].equals("/to")) {
+                state = 2;
+            } else {
+                if (state == 1) {
+                    from += inputComponent[i];
+                    from += " ";
+                } else if (state == 2) {
+                    to += inputComponent[i];
+                    to += " ";
+                } else {
+                    description += inputComponent[i];
+                    description += " ";
+                }
+            }
+        }
+
+
+        taskList[taskCount] = new Event(description.trim(), from.trim(), to.trim());
+        System.out.println("Got it. I've added this task:");
+        System.out.println(taskList[taskCount].toString());
+        taskCount++;
+        System.out.println("Now you have " + taskCount + " tasks in the list");
+    }
+
+
+    //main function to execute the chatbot
+    public void execute() {
+        System.out.println("Hello I'm Lambo");
+        System.out.println("What can I do for you?");
+        Scanner inputReader = new Scanner(System.in);//scanner for receiving input
+
+
+        //Super loop used for receiving inputs continuously and response back to user
+        SuperLoop:
+        while (true) {
+            String input = inputReader.nextLine();
+            String[] inputComponent = input.split(" ");
```

Comment:
Perhaps rename `inputComponent` to `inputComponents` (note the plural form) to better indicate its status as an array of Strings?